### DISCUSSION TITLE
KeyError Message : File "C:\Users\IT UW\odoo14\flectra\flectra\fields.py", line 509, in _setup_related_full     field = target._fields[name] KeyError: 'type_kk' - - -

### DISCUSSION THREAD

**Discussion by it12uw** [OCA/odoo-community.org#98] [General]
hello guys, i need help to fix my problem.
here is my problem, i have model instaled on the system, but i just update the model and every time and try to update the model
i always got this error message  **File "C:\Users\IT UW\odoo14\flectra\flectra\fields.py", line 509, in _setup_related_full
    field = target._fields[name]
KeyError: 'type_kk' - - -**

here is the xml code

<?xml version="1.0" encoding="UTF-8"?>
<flectra>
    <data>
        <record model="ir.ui.view" id="view_product_product_form">
        <field name="name">product.product.form</field>
        <field name="model">product.product</field>
        <field name="inherit_id" ref="product.product_normal_form_view"/>
        <field name="type">form</field>
        <field name="arch" type="xml">
                <xpath expr="//page[@name='general_information']" position="after">
                    <page string="Data Barang Produksi Produk" name="data_barang_produksi_produk">
                        <group>
                            <group name="group_general">
                                <field name="nama_desain"/>
                                <field name="no_mesin"/>
                                <field name="berat"/>
                                <field name="warna"/>
                                <field name="type_kk"/>
                                <field name="tgl_buat"/>
                            </group>

                            <group name="group_data_material">
                                <group name="main_grp_1" colspan="2">
                                    <group name="main_grp_col_1">
                                        <field name="gum_stretch_x"/>
                                        <field name="gum_stretch_y"/>
                                        <field name="gum_relaxed_x"/>
                                        <field name="gum_relaxed_y"/>
                                    </group>
                                    <group name="main_grp_col1">
                                        <field name="leg_gum_stretch_x"/>
                                        <field name="leg_gum_stretch_y"/>
                                        <field name="leg_gum_relaxed_x"/>
                                        <field name="leg_gum_relaxed_y"/>
                                    </group>
                                </group>

                                <group name="main_grp_2" colspan="2">
                                    <group name="main_grp_col_2">
                                        <field name="leg_stretch_x"/>
                                        <field name="leg_stretch_y"/>
                                        <field name="leg_relaxed_x"/>
                                        <field name="leg_relaxed_y"/>
                                    </group>
                                    <group name="main_grp_col2">
                                        <field name="foot_stretch_x"/>
                                        <field name="foot_stretch_y"/>
                                        <field name="foot_relaxed_x"/>
                                        <field name="foot_relaxed_y"/>
                                    </group>
                                </group>

                                <group name="main_grp_3" colspan="2">
                                    <group name="main_grp_col_3">
                                        <field name="foot_gum_stretch_x"/>
                                        <field name="foot_gum_stretch_y"/>
                                        <field name="foot_gum_relaxed_x"/>
                                        <field name="foot_gum_relaxed_y"/>
                                    </group>
                                    <group name="main_grp_col3">
                                        <field name="hell_stretch_x"/>
                                        <field name="hell_stretch_y"/>
                                        <field name="hell_relaxed_x"/>
                                        <field name="hell_relaxed_y"/>
                                    </group>
                                </group>
                            </group>
                        </group>
                    </page>

                    <page string="Data Bahan Produksi" name="data_bahan_produksi">
                        <group>
                            <field name="kode_benang"/>
                            <field name="no_warna_benang"/>
                            <field name="jenis_benang"/>
                            <field name="warna_benang"/>
                            <field name="size"/>
                            <field name="style"/>
                            <field name="kategoti"/>
                        </group>
                    </page>
                </xpath>
            </field>
        </record>
    </data>
</flectra>`

and this is the py code

`
from flectra import api, fields, models, _
from datetime import datetime

class InMrpProductProduct(models.Model):
    _name = 'product.product'
    _inherit = 'product.product'    

    #product
    nama_desain = fields.Char(string='Nama Desain',)
    no_mesin= fields.Float(string='No Mesin/Jarum',)
    berat= fields.Float(string='Berat/Pasang',)
    warna = fields.Char(string='Warna',required=True, default='NA')
    tgl_buat = fields.Date(string='tgl Buat',default=fields.Date.context_today,readonly=True,)
    style = fields.Char(string=u'Style',required=True, default='NA')
    
    gum_stretch_x= fields.Float(string='Gum Stretch x ',)
    gum_stretch_y= fields.Float(string='Gum Stretch y ',)
    leg_gum_stretch_x= fields.Float(string='Leg Gum Stretch x ',)
    leg_gum_stretch_y= fields.Float(string='Leg Gum Stretch y ',)
    leg_stretch_x= fields.Float(string=' Leg Stretch x',)
    leg_stretch_y= fields.Float(string='Leg Stretch y ',)
    foot_stretch_x= fields.Float(string=' Foot Stretch x',)
    foot_stretch_y= fields.Float(string=' Foot Stretch y',)
    foot_gum_stretch_x= fields.Float(string='Foot Gum Stretch x ',)
    foot_gum_stretch_y= fields.Float(string='Foot Gum Stretch y ',)
    hell_stretch_x= fields.Float(string='Hell Stretch x ',)
    hell_stretch_y= fields.Float(string='Hell Stretch y ',)

    gum_relaxed_x= fields.Float(string='Gum Relaxed x ',)
    gum_relaxed_y= fields.Float(string='Gum Relaxed y ',)
    leg_gum_relaxed_x= fields.Float(string='Leg Gum Relaxed x ',)
    leg_gum_relaxed_y= fields.Float(string='Leg Gum Relaxed y ',)
    leg_relaxed_x= fields.Float(string=' Leg Relaxed x',)
    leg_relaxed_y= fields.Float(string='Leg Relaxed y ',)
    foot_relaxed_x= fields.Float(string=' Foot Relaxed x',)
    foot_relaxed_y= fields.Float(string=' Foot Relaxed y',)
    foot_gum_relaxed_x= fields.Float(string='Foot Gum Relaxed x ',)
    foot_gum_relaxed_y= fields.Float(string='Foot Gum Relaxed y ',)
    hell_relaxed_x= fields.Float(string='Hell Relaxed x ',)
    hell_relaxed_y= fields.Float(string='Hell Relaxed y ',)

    #bahan baku    
    kode_benang = fields.Char(string='Kode Benang',)
    no_warna_benang = fields.Char(string='No Warna Benang',)
    warna_benang = fields.Char(string='Warna Benang',)
    jenis_benang = fields.Char(string='Jenis Benang',)
    
    #---- 
    size = fields.Char(string='Size Steam',required=True, default='NA')
    kategoti = fields.Text(string=u'Kategori',)

    type_kk = fields.Selection([('I', 'I'),('S', "S"),('-', "-")],string='Type KK',)

  `

can you guys help me to fix this problem?
thankyou :)

**Comment by etobella** [comment]
You are using flectra with odoo?

**Reply by it12uw** [reply]
yes my company using flectra with odoo

**Reply by pedrobaeza** [reply]
Then it's normal that is not working. Flectra is a fork of Odoo, and compatibility is not guaranteed between both. Get a special version of the module adapted to Flectra, or switch to Odoo.

**Reply by it12uw** [reply]
how to switch from flectra to odoo?

**Reply by legalsylvain** [reply]
> how to switch from flectra to odoo?

no idea. Did you try to put your flectra database / filestore in an Odoo 11 instance ?
